extensions/README.md

```diff
@@ -0,0 +1,212 @@
+
+Extensions are controled via:
+* the ``DIRACX_EXTENSIONS`` environment variable. It should be set to `gubbins,diracx`
+* the entrypoints found in the various `pyproject.toml`
+
+This here is the exhaustive list of what is supported and tested for extensions. Any usecase not listed here is not supported.
+
+
+## QUESTIONS
+
+What to do with the `environment.yaml` ? should we replicate wht's in diracx ?
+
+
+## General statements
+
+When working with the extension, you MUST always (server, client, testing) have exported the environment variable ``DIRACX_EXTENSIONS=gubbins,diracx``. The [``pytest fixture``](extensions/gubbins/gubbins-testing/src/gubbins/testing/__init__.py)
+
+
+The fact of having `gubbins` as a subfolder has a few downside which you will not suffer if having your extension in a separate repository:
+* the `root` of ``setuptools_scm`` in the various ``pyproject.toml`` will only be ``..`` for your extension
+
+## CI
+
+The extension is tested in the CI.
+
+What is in the [action file](``.github/workflows/extensions.yaml``) should in fact be split in multiple jobs under ``.github/workflows/`` of your repo.
+
+Here we use the local versions of `diracx` packages to make sure we are always up to date, and that a change in `diracx` does not break the extension mechanisms. But in your real extension, you will want to depend on published package (i.e. ``pip install diracx-routers`` instead of ``pip install -e ./diracx-routers`` for example), and on published docker images.
+
+Moreover, the `gubbins` docker images are never uploaded, and are only passed from on job to the next. You should definitely upload yours.
+
+## Installing the extension
+
+To develop, follow the same instruction as `diracx`
+
+```bash
+# Create the mamba environment
+mamba env create --file extensions/gubbins/environment.yml
+conda activate gubbins-dev
+
+
+# Make an editable installation of diracx
+pip install -r requirements-dev.txt
+
+# Make an editable installation of gubbins
+cd extensions/gubbins/
+pip install -r requirements-dev.txt
+
+# Install the patched DIRAC version
+pip install git+https://github.com/DIRACGrid/DIRAC.git@integration
+
+# Enable pre-commit
+mamba install pre-commit
+pre-commit install
+```
+
+## DB
+
+The ``gubbins-db`` package contains the extension for the DB.
+
+### New DB
+
+`lollygag` is a DB which is specific to `gubbins`, i.e. it does not modify or extend an existing `diracx` db
+
+### Extended DB
+
+`GubbinsJobDB` illustrate how to extend an existing `diracx` DB, add new methods, modify methods, add a table.
+
+A [router test](extensions/gubbins/gubbins-routers/tests/test_gubbins_job_manager.py) exists, even though no router is redefined. It is just to show that the correct DB is being loaded.
+
+> [!WARNING]
+> in the dependency override, you need to specify both the original DiracX JobDB as well as the extended one GubbinsJobDB. To avoid these issues, reuse the same name (i.e. `JobDB` instead of `GubbinsJobDB`)
```

Comment:
It is not crystal clear to me:
- does "dependency override" mean `<extension>-db/pyproject.toml`?
- why do I need and how can I specify both the original JobDB and the extended one?
- "To avoid these issues", which ones? (Sorry I am reading that fast, so I might have overlooked something)

Reply:
That here: https://github.com/chaen/diracx/blob/63e9f87d56d0f65da9669d4c33a3802483505b1f/extensions/gubbins/gubbins-routers/tests/test_gubbins_job_manager.py#L9-L23

I'll rephrase